main.py

```diff
@@ -7,41 +17,66 @@ def get_arguments(arguments):
     number_of_arguments = len(arguments)
     print("Number of arguments", number_of_arguments)
 
-    if number_of_arguments <= 1 or number_of_arguments > 2:
+    if number_of_arguments < 1 or number_of_arguments > 2:
         print("Incorrect command usage, please follow the correct format")
         print("COMMAND: python3 [link] [max_repos]")
-        return []
+        return command_failed()
     return arguments
 
 
 def validate_arguments(arguments):
-    if len(arguments) == 0:
-        return
+    if arguments is None:
+        return command_failed()
 
+    first_argument = arguments[0]
     second_argument = arguments[1]
```

Comment:
2nd argument is optional, so arguments here may only be 1 item. Then this line will cause the code to crash. Check length before accessing 2nd item

Reply:
made the 2nd argument optional [3e8a47d]